### PR TITLE
 delete resource script added

### PR DESCRIPTION
**Pull request by bergano65**

### CODE REVIEW

File: CogsMinimizer/Views/Subscription/Analyze.cshtml [hunk 1]
```diff
@@ -306,6 +310,52 @@
...
+            url: "/Subscription/GetResourceForDelete",
+            type: "POST",
+            dataType: "json",
+            data: JSON.stringify({
+                SubscriptionId: '@Model.SubscriptionData.Id'
+            }),
+            contentType: "application/json; charset=utf-8",
+            success: function (data) {
+
+                var content = 'Write-host \'Continue will delete resources listed in script. Are you Sure You Want To Proceed? (y/n)\':\n' +
```
Comment (shklar) [inline CogsMinimizer/Views/Subscription/Analyze.cshtml]: Running this script will permanently delete all the resources in the subscription that have been identified as obsolete using SubMinimizer.  
Subscription name: <subscription name> 
Subscription id: <subscription id>
Are you sure? Y/N

File: CogsMinimizer/Views/Subscription/Analyze.cshtml [hunk 2]
```diff
@@ -306,6 +310,52 @@
...
+                    '$confirm = Read-Host\n' +
+                    'if ($confirm -ne \'y\')\n' +
+                    '{\n' +
+                'return\n' +
+            '}\n' +
+                'Connect-AzureRmAccount\n' +
```
Comment (shklar) [inline CogsMinimizer/Views/Subscription/Analyze.cshtml]: What happens if the user does not have access to the subscription?

File: CogsMinimizer/Views/Subscription/Analyze.cshtml [hunk 3]
```diff
@@ -306,6 +310,52 @@
...
+                'Set-AzureRmContext -Subscription \'{0}\'\n';
+
+                content = content.replace('{0}', '@Model.SubscriptionData.Id');
+
+                for (var resNum = 0; resNum < data.length; resNum++) {
+                    var resource = data[resNum];
+
```
Comment (shklar) [inline CogsMinimizer/Views/Subscription/Analyze.cshtml]: For each resource please add a PowerShell comment saying:
"Deleting resource <resource-name> from resource group <resource-group>. Last claimed by <owner -name or unclaimed>

File: CogsMinimizer/Views/Subscription/Analyze.cshtml [hunk 4]
```diff
@@ -306,6 +310,52 @@
...
+                content = content.replace('{0}', '@Model.SubscriptionData.Id');
+
+                for (var resNum = 0; resNum < data.length; resNum++) {
+                    var resource = data[resNum];
+            
+                    content += "Remove-AzureRmResource -Force -ResourceId '" + resource.AzureId + "'\n";
+                }                        
+
+                var fileName = 'DelResources.ps1';
```
Comment (shklar) [inline CogsMinimizer/Views/Subscription/Analyze.cshtml]: Please make the file name specific with subscription name and date:
DeleteObsoleteResource_<sub_name>_<DD-MM-YY>.ps1

File: CogsMinimizer/Views/Subscription/Analyze.cshtml [hunk 5]
```diff
@@ -306,6 +310,52 @@
...
+                var blob = new Blob([content],
```
Comment (shklar) [inline CogsMinimizer/Views/Subscription/Analyze.cshtml]: Do these blobs get stored? We need to make sure they are generated only if needed, and deleted afterwards.

File: CogsMinimizer/Views/Subscription/Analyze.cshtml [hunk 6]
```diff
@@ -618,9 +668,12 @@
                     {
                        <div class="row">
                             <button class="btn btn-warning btn-round btn-cmd" title="Makes all resources unclaimed. Sets expiration date to established for unclaimed resources." onclick="OnResetResources()">Reset resources</button>
-                            <br />
                         </div>
                     }
+                    <div class="row">
+                        <button style="word-wrap: break-word;" class="btn btn-warning btn-round btn-cmd" title="Download delete resources script" onclick="OnGetDeleteResourcesScript()">Download delete<br/>resources script</button>
```
Comment (shklar) [inline CogsMinimizer/Views/Subscription/Analyze.cshtml]: Do this in a single line (no need for the <br/> in the middle)

File: CogsMinimizer/Controllers/SubscriptionController.cs [hunk 7]
```diff
@@ -19,6 +19,49 @@ public class SubscriptionController : SubMinimizerController
...
+            {
+                var subscription = db.Subscriptions.FirstOrDefault(x => x.Id.Equals(SubscriptionId));
+
+                if (subscription == null)
+                {
+                    throw new ArgumentException(string.Format("Subscription with ID '{0}' wasn't found.", SubscriptionId));
+                }
+                
+                // Let's compose result list in order to build resources delete script
+                List<Resource> subscriptionResourceList = db.Resources.Where(r => r.SubscriptionId == SubscriptionId).ToList();
```
Comment (shklar) [inline CogsMinimizer/Controllers/SubscriptionController.cs]: Why not add the filter for "Marked for delete" resources directly to the DB query?

File: CogsMinimizer/Controllers/SubscriptionController.cs [hunk 8]
```diff
@@ -19,6 +19,49 @@ public class SubscriptionController : SubMinimizerController
     {
         public const int EXPIRATION_INTERVAL_IN_DAYS = 7;
 
+        // Get resources for deleting
+        [HttpPost]
+        public ActionResult GetResourceForDelete(string SubscriptionId)
```
Comment (shklar) [inline CogsMinimizer/Controllers/SubscriptionController.cs]: Not sure why we need the additional controller.
Don't we have all the information in the ViewModel of the Analyze view?

File: CogsMinimizer/Views/Subscription/Analyze.cshtml (hunk 2 above)
Comment (bergano65) [reply CogsMinimizer/Views/Subscription/Analyze.cshtml]: you mean at time of script running?  depending from your rights you will be thrown at setting  current  subscription if  forbidden reading. or at deleting if forbidden writing.  anyway  connect account  requires login and at continue check rights

File: CogsMinimizer/Views/Subscription/Analyze.cshtml (hunk 1 above)
Comment (bergano65) [reply CogsMinimizer/Views/Subscription/Analyze.cshtml]: fixed

File: CogsMinimizer/Views/Subscription/Analyze.cshtml (hunk 3 above)
Comment (bergano65) [reply CogsMinimizer/Views/Subscription/Analyze.cshtml]: fixed

File: CogsMinimizer/Views/Subscription/Analyze.cshtml (hunk 4 above)
Comment (bergano65) [reply CogsMinimizer/Views/Subscription/Analyze.cshtml]: fixed

File: CogsMinimizer/Views/Subscription/Analyze.cshtml (hunk 6 above)
Comment (bergano65) [reply CogsMinimizer/Views/Subscription/Analyze.cshtml]: fixed

File: CogsMinimizer/Controllers/SubscriptionController.cs (hunk 7 above)
Comment (bergano65) [reply CogsMinimizer/Controllers/SubscriptionController.cs]: fixed

File: CogsMinimizer/Controllers/SubscriptionController.cs (hunk 8 above)
Comment (shklar) [reply CogsMinimizer/Controllers/SubscriptionController.cs]: If this is tricky then just leave it as is.

Comment (bergano65) [reply CogsMinimizer/Controllers/SubscriptionController.cs]: fixed

**Pull request closed without merging**